### DISCUSSION TITLE
[Bug]: Write permission for FSL

### DISCUSSION THREAD

**Discussion by navonacalarco** [neurodesk/neurodesk.github.io-hugo#408] [Bug Report]
### Version

1.4.0

### What platform are you seeing the problem on?

Local PC

### What happened

Hello! I am pretty sure this is a user error and not a bug, but, I can't figure out how to get FSL run via Neurodesk to make directories or write files. 

An example command is `run_first_all -b -d -m fast  -i subject_acq-uni_final_template0.nii.gz -o subject_fslSegFile.nii.gz` 

In both cases, I get `Permission denied` issues, e.g.:
`mkdir: cannot create directory '/home/jovyan/data/storage/......./subject_fslSegFile.logs': Permission denied`
`Error submitting job - Unable to create /home/jovyan/data/storage/.../subject_fslSegFile.logs ([Errno 13]`

Both directories in which files would be made/written have full permissions. The same code works on the local install of FSL when run outside of Neurodesk, on the same computer.

Apologies for such a basic question - but is there any special trick to allowing Neurodesk to create/write files?

### How to reproduce

FSL v 6.0.7.4

**Comment by stebo85** [comment]
Dear @navonacalarco - it sounds like the permissions of the directory are not quite right. To find out what’s happening:

can you open a standard terminal (so not an fsl terminal) in neurodesk and navigate to the directory. Do you have write permissions there?

if yes: you need to add the directory to the singularity bind path variable

if no: you need to set the permissions to writable for all users in your host operating system 

is your host operating system Linux or macOS? Are you starting neurodesk through the Neurodeskapp or through the docker command on the terminal? If you are starting it through a terminal command can you send this command as well?

**Comment by navonacalarco** [comment]
Dear @stebo85 - thank you so much! It was indeed a permissions error; changing the permissions for the group (cf. owner) fixed it. Apologies, not sure how I missed that before. Thanks again for this software. :)

**Reply by stebo85** [reply]
Dear @navonacalarco - thank you for letting us know that this worked. @Vbitz had an idea of how we could fix the permissions on startup automatically. Once he documented this fix on our website I hope we can prevent this problem for people entirely.

**Comment by navonacalarco** [comment]
[heart]         Navona Calarco reacted to your message:
________________________________
